app/src/main/java/com/celzero/bravedns/receiver/DeviceLockService.kt

```diff
@@ -52,35 +52,30 @@ class DeviceLockService  : Service(){
 
         val isProtected = keyguardManager.isKeyguardSecure
         val isLocked = keyguardManager.isKeyguardLocked
+        // ref - https://developer.android.com/reference/android/os/PowerManager#isInteractive()
         val isInteractive = powerManager.isInteractive
         val delayIndex: Int = getSafeCheckLockDelay(intent.getIntExtra(EXTRA_CHECK_LOCK_DELAY_INDEX, -1))
 
-
-        if (checkLockTask != null) {
-            checkLockTask!!.cancel()
-        }
+        checkLockTask?.cancel()
 
         if (isProtected && !isLocked && !isInteractive) {
             checkLockTask = CheckLockTask(this, delayIndex)
             timer.schedule(checkLockTask, checkLockDelays[delayIndex].toLong())
             this.stopSelf()
         } else {
-            if (isProtected && isLocked) {
-                if (persistentState.getFirewallModeForScreenState() && !persistentState.getScreenLockData()) {
-                    if(DEBUG) Log.d(LOG_TAG,"DeviceLockService : Screen lock detected at $delayIndex")
-                    persistentState.setScreenLockData(true)
-                    checkLockTask?.cancel()
-                    timer.cancel()
-                    this.stopSelf()
-                }
+            if (!isProtected || !isLocked) return
+            if (!persistentState.isScreenOff) {
```

Comment:
The shared-preference is one of the best way to store and access the application settings, other than that there are very few toggle persistent-state which need to be removed. Created an issue for the isScreenOff earlier #304

Reply:
> The shared-preference is one of the best way

The problem isn't with shared-preferences, but the way it is used. It is not transactional (as in, updating values to it may not necessarily reflect in other parts of the code), and so, with the app-state and the vpn-state spread across multiple shared-preference variables, this is bound to enter into all sorts of race conditions.

Reply:
`DeviceLockService` has been removed in its entirety.